src/test/java/uk/gov/hmcts/reform/idam/web/AppControllerTest.java

```diff
@@ -1509,4 +1518,32 @@ public void makeCookieSecure_shouldReturnANonsecureCookieIfUseSecureCookieIsFals
         assertThat(appController.makeCookieSecure(AUTHENTICATE_SESSION_COOKE, false), is(AUTHENTICATE_SESSION_COOKE + "; Path=/; HttpOnly"));
     }
 
+    /**
+     * @verifies put in model the correct error variable in case policy check fails
+     * @see AppController#login(AuthorizeRequest, BindingResult, Model, HttpServletRequest, HttpServletResponse)
+     */
+    @Test
+    public void login_shouldPutInModelTheCorrectErrorVariableInCasePolicyCheckFails() throws Exception {
+        given(spiService.authenticate(eq(USER_EMAIL), eq(USER_PASSWORD), eq(USER_IP_ADDRESS))).willReturn(AUTHENTICATE_SESSION_COOKE);
+        given(spiService.authorize(any(), eq(AUTHENTICATE_SESSION_COOKE))).willReturn(REDIRECT_URI);
+        given(policyService.evaluatePoliciesForUser(any(), any(), any()))
+            .willReturn(false);
+
+        mockMvc.perform(post(LOGIN_ENDPOINT).with(csrf())
+            .header(X_FORWARDED_FOR, USER_IP_ADDRESS)
+            .param(USERNAME_PARAMETER, USER_EMAIL)
+            .param(PASSWORD_PARAMETER, USER_PASSWORD)
+            .param(REDIRECT_URI, REDIRECT_URI)
+            .param(STATE_PARAMETER, STATE)
+            .param(RESPONSE_TYPE_PARAMETER, RESPONSE_TYPE)
+            .param(CLIENT_ID_PARAMETER, CLIENT_ID)
+            .param(SCOPE_PARAMETER, CUSTOM_SCOPE))
+            .andExpect(status().isOk())
+            .andExpect(model().attribute(MvcKeys.HAS_POLICY_CHECK_FAILED, true))
+            .andExpect(view().name(LOGIN_VIEW));
+
+        verify(spiService).authorize(any(), eq(AUTHENTICATE_SESSION_COOKE));
+
+        verify(policyService).evaluatePoliciesForUser(eq(REDIRECT_URI), eq(AUTHENTICATE_SESSION_COOKE), eq(USER_IP_ADDRESS));
+    }
```

Comment:
I guess there's an existing test for failed login - might be worth have a `verified(policyService..., never()...` on there.